### DISCUSSION TITLE
Help with multiplexers.

### DISCUSSION THREAD

**Discussion by kilehh** [tttapa/Control-Surface#514] [Q&A] [answered]
Hello!
I have multiplexer CD74HC4067, connected to A0, 14, 15, 16, 17 pins, to multiplexer i have connected 15 potentiometers. I am using arduino mega and for midi interface i want to use hairlessMIDI.  
I looked at examples and used code from there like this, and its not showing up in hairlessMIDI.

Code:
~~~cpp
#include <Control_Surface.h>

HairlessMIDI_Interface midi;

CD74HC4067 mux {
  A0, 
  {14, 15, 16, 17} };

CCPotentiometer volumePotentiometers[] {
  {mux.pin(0), {MIDI_CC::Channel_Volume, CHANNEL_1}},
  {mux.pin(1), {MIDI_CC::Channel_Volume, CHANNEL_2}},
  {mux.pin(2), {MIDI_CC::Channel_Volume, CHANNEL_3}},
  {mux.pin(3), {MIDI_CC::Channel_Volume, CHANNEL_4}},
  {mux.pin(4), {MIDI_CC::Channel_Volume, CHANNEL_5}},
  {mux.pin(5), {MIDI_CC::Channel_Volume, CHANNEL_6}},
  {mux.pin(6), {MIDI_CC::Channel_Volume, CHANNEL_7}},
  {mux.pin(7), {MIDI_CC::Channel_Volume, CHANNEL_8}},
  {mux.pin(8), {MIDI_CC::Channel_Volume, CHANNEL_9}},
  {mux.pin(9), {MIDI_CC::Channel_Volume, CHANNEL_10}},
  {mux.pin(10), {MIDI_CC::Channel_Volume, CHANNEL_11}},
  {mux.pin(11), {MIDI_CC::Channel_Volume, CHANNEL_12}},
  {mux.pin(12), {MIDI_CC::Channel_Volume, CHANNEL_13}},
  {mux.pin(13), {MIDI_CC::Channel_Volume, CHANNEL_14}},
  {mux.pin(14), {MIDI_CC::Channel_Volume, CHANNEL_15}},
  {mux.pin(15), {MIDI_CC::Channel_Volume, CHANNEL_16}},
};

void setup() {
  Control_Surface.begin();
}

void loop() {
  Control_Surface.loop();
}
~~~

**Comment by tttapa** [comment, marked as answer]
What is not showing up in Hairless? The port or the MIDI messages?

**Reply by kilehh** [reply]
Midi messages

**Reply by tttapa** [reply]
You'll have to provide more information. Are you sure you selected the right serial port? Did you close the Arduino IDE/Serial monitor before opening Hairless? What baud rate did you select in the Hairless settings (it should be 115200)?

**Reply by kilehh** [reply]
Okay the problem was that boud rate wasnt set at 115200, and i did change it, but now only half of the potentiometers work. I did found out that if i move one of the potentiometers in hairless it says "Serial in Ch 4: Controller 7 value 20" and "Serial in Ch 2: Controller 7 value 20" and i dont know if thats how it should be. Also some of the potentiometers have high noise which couses loopbe1 to show up a error and mute, and that also happens if i move the knobs too fast and alot. 
What should i do?

**Reply by tttapa** [reply]
Problems with half of the potentiometers are often caused by poor connection of one of the address lines of the multiplexer. Excessive noise is a hardware problem as well.

**Reply by kilehh** [reply]
After few nights of troubleshooting i found out that the problem was the wires or the mux it self. I think that the problem could have been the wires, because they were of poor quality, broke off easily and had other problems (and yes, i did check the soldering and connection was good), since i was resoldering them, i also replaced the mux, because i had another one laying around. All that fixed my issue. 
Big thanks for your help and leading me in the right direction!
